### PR TITLE
Add API token authentication scheme

### PR DESCRIPTION
**Pull request by super-cooper**
- Create new authentication models and migrations
- Remove redundant FlotoAuthBackend
- Add test for API token authentication
- Add interface for users to obtain API keys

### CODE REVIEW

File: floto/settings.py [hunk 1]
```diff
@@ -87,12 +92,24 @@
 # Database
 # https://docs.djangoproject.com/en/4.1/ref/settings/#databases
 
-DATABASES = {
-    'default': {
-        'ENGINE': 'django.db.backends.sqlite3',
-        'NAME': BASE_DIR / 'db.sqlite3',
+if os.getenv("DB_ENGINE"):
```
Comment (Mark-Powers) [inline floto/settings.py:95]: thank you! I didn't commit this yet, my apologies

**Review by Mark-Powers — APPROVED** [review]
This looks great! I have some questions about how the user model works. The Mozilla OIDC backend automatically calls on this when user logs in for the first time? It looks to me like this info is cached per session, right? And so we don't have to worry about updating the user DB row. I recall in Chameleon Portal, there is some updating code.

I think this will also solves some issues (or can be used to) regarding checking the user permissions.

**Comment by super-cooper** [comment]
> The Mozilla OIDC backend automatically calls on this when user logs in for the first time?

Correct

We might want to add some code to update on existing users. I think if their email is updated in keycloak, for example, it remains the same in the dashboard.

File: floto/settings.py (hunk 1 above)
Comment (super-cooper) [reply floto/settings.py:95]: No worries :)

**Pull request merged**